apis/projectcontour/v1/httpproxy.go

```diff
@@ -391,6 +449,27 @@ type Route struct {
 	// DirectResponsePolicy returns an arbitrary HTTP response directly.
 	// +optional
 	DirectResponsePolicy *HTTPDirectResponsePolicy `json:"directResponsePolicy,omitempty"`
+
+	// The policy for verifying JWTs for requests to this route.
+	// +optional
+	JWTVerificationPolicy *JWTVerificationPolicy `json:"jwtVerificationPolicy,omitempty"`
+}
+
+type JWTVerificationPolicy struct {
+	// Require names a specific JWT provider (defined in the virtual host)
+	// to require for the route. If specified, this field overrides the
+	// default provider if one exists. If this field is not specified,
+	// the default provider will be required if one exists. At most one of
+	// this field or the "disabled" field can be specified.
+	// +optional
+	Require string `json:"require,omitempty"`
```

Comment:
thinking out loud: we could pull this out into a struct to make it easier in the future to add more a advanced requirement pattern, but i think that would still maybe require deprecating fields so probably not worth it, just a small thing to think about again in case anyone finds it useful

Reply:
I had been thinking that the `JWTVerificationPolicy` struct itself could add `RequiresAll`, `RequiresAny`, etc. fields down the road if needed, with only one of them being allowed to be specified but yeah we could also encapsulate all of those into a separate struct